tests/test_sheet_collector.py

```diff
@@ -395,3 +396,21 @@ def test_sheet_collector_collect_files_prints_output(tmpdir, test_data, capfd):
     captured = capfd.readouterr()
     for print_item in test_data["collect_files_test"]["expected_print"]:
         assert print_item in captured.out
+
+
+def test_grab_returns_actual_url():
+    """Test that function returns correct ID from full URL."""
+    expected_id = "1XKnoa1BBzEnJ1TA_LTRs5e0zcva0SCgNyt7cfMVGHWc"
+    url_one = util.extract_sheet_id(
+        "https://docs.google.com/spreadsheets/d/1XKnoa1BBzEnJ1TA_LTRs5e0zcva0SCgNyt7cfMVGHWc/edit"
+    )
+    assert url_one == expected_id
+
+
+def test_grab_returns_no_prefix():
+    """Test that function returns correct ID from partial URL."""
+    expected_id = "my_very_important_key"
+    url_two = util.extract_sheet_id(
+        "docs.google.com/spreadsheets/d/my_very_important_key/edit#gid=0"
+    )
+    assert url_two == expected_id
```

Comment:
Nice start to test cases, but there is more things to add. Here are things to consider:
- test empty strings
- test an invalid URL and check that an exception is being raised (pytest has a nice way to do that)
- test what happens when there is no `/d` in the URL
- Think about other cases to test here

Reply:
I've messaged @ColeKaupelis to take a look at this--one of us will implement the add'l suggestions before the week's out.

Reply:
Any updates on testing @jnormile ?